kratos/includes/variables.h

```diff
@@ -47,6 +47,7 @@ namespace Kratos
 
     //ints
     KRATOS_DEFINE_VARIABLE( int, DOMAIN_SIZE )
+    KRATOS_DEFINE_VARIABLE( int, DIMENSION )
```

Comment:
I strongly suggest giving this var another name as we have plenty of non-kratos variables and enums named `DIMENSION` in other apps.

Reply:
Well this is a  Kratos Variable used for other purposes. If there is any conflict in the future it can be changed.

Reply:
The problem is that if we left it this way it does conflict, just the fact that it exists is a problem. For example, in DEM application there are multiple enums that define DIMENSION as a integer ( this one translates as a KratosVariable<int>). And probably it is not a easy to detect problem when/if it happens. 

Specifically what I amb afraid of is that some kind of partial shadowing could happend depending on which files are included or not. For example consider this code:

```C++
#include "variables.h"
#include "myapp.h" #but you don't include the file with the enum {DIMENSION = 3}

for(int i = 0; i < DIMENSION; i++) {
    doSomething();
}
```

This code probably compiles, and maybe only loop once, maybe it crashes, maybe loops 4 times and we end up geting a memleak.

Another solution would be to move it to your application or to pfem base / contact emchanichs if it needs to be inherited by anyone else. This way is isolated from other people, then the problem is only while coupling with other applications, which is perfectly fine.

Reply:
Ok I see, the problem is critical and I have new questions:
1) This happens because these enums write the variables in CAPITALS which coincide with Kratos Variables standard ?  If the enum name is "Dimension" then no problem occurs?
(if it is like this, would be convienient to not use capitals for enum variables ?)
2) If a Kratos Variable and a Global or Local flag have the same name (they are usually written in capitals), have we got the same problem?
3) I think it is planned (in the future) to define the variables in the applications, not in Kratos core, then variables could be repeated. However it will be also dangerous because we can find the same variable defined as a local variable inside some classes.
I think you must expose this problem to @KratosMultiphysics/technical-committee .

Right now I will change the name to SPACE_DIMENSION to avoid conflicts and I will add the correction to this branch.

Reply:
1) Yes, that is the problem indeed. As you said it won't be a problem if DIMENSION enum or C++ variables would have been called "Dimension" without caps, but enums usually are written in caps to show that are meaningful values.
2) Yes, if you are aware of any flag where this happens pls tell me.
3) Again, you are right. While in general it should not be the case this may happen. Probably the problem with dimension is that is somehow a very "generic" word. I recall having similar issues with VELOCITY for example. We will for sure discuss this in the next @KratosMultiphysics/technical-committee meeting.